osquery/tables/system/darwin/disk_encryption.mm

```diff
@@ -0,0 +1,347 @@
+    if (apfs == nullptr) {
+      LOG(ERROR) << "Could not allocate DMAPFS object";
+      CFRelease(session);
+      CFBundleUnloadExecutable(bundle);
+      CFRelease(bundle);
+      return;
+    }
+    [apfs performSelector:@selector(initWithManager:) withObject:man];
+
+#pragma clang diagnostic pop
+
+    DADiskRef targetVol = DADiskCreateFromBSDName(nullptr, session, r["name"].c_str());
+
+    // err = [apfs isEncryptedVolume:targetVol encrypted:&isEncrypted];
+    char *typeEncodings;
+    asprintf(&typeEncodings,
+             "%s%s%s%s%s",
+             @encode(int), // return
+             @encode(id), // self
+             @encode(SEL), // _cmd
+             @encode(DADiskRef),
+             @encode(char*));
+    NSMethodSignature *signature = [NSMethodSignature signatureWithObjCTypes: typeEncodings];
+    free(typeEncodings);
+
+    char isEncrypted = 1;
+    char *isEncryptedPtr = &isEncrypted;
+    int err = 0;
+
+    NSInvocation *inv = [NSInvocation invocationWithMethodSignature:signature];
+    [inv setSelector:@selector(isEncryptedVolume:encrypted:)];
+    [inv setReturnValue:&err];
+    [inv setArgument:&targetVol atIndex:2];
+    [inv setArgument:&isEncryptedPtr atIndex:3];
+    [inv invokeWithTarget:apfs];
+    if (targetVol == nullptr) {
+      LOG(ERROR)
+          << "Error calling isEncryptedVolume:encrypted:, targetVol is null";
+      CFRelease((__bridge CFTypeRef)apfs);
+      CFRelease(session);
+      CFBundleUnloadExecutable(bundle);
+      CFRelease(bundle);
+      return;
+    }
+    if (err != 0) {
+      LOG(ERROR)
+          << "Error calling isEncryptedVolume:encrypted:, targetVol not null";
+      CFRelease(targetVol);
+      CFRelease((__bridge CFTypeRef)apfs);
+      CFRelease(session);
+      CFBundleUnloadExecutable(bundle);
+      CFRelease(bundle);
+      return;
+    }
+
+    // err = [apfs cryptoUsersForVolume:targetVol users:&cryptoUsers];
+    asprintf(&typeEncodings,
+             "%s%s%s%s%s",
+             @encode(int), // return
+             @encode(id), // self
+             @encode(SEL), // _cmd
+             @encode(DADiskRef),
+             @encode(void*));
+    signature = [NSMethodSignature signatureWithObjCTypes: typeEncodings];
+    free(typeEncodings);
+
+    NSArray *cryptoUsers;
+    void *cryptoUsersPtr = &cryptoUsers;
+    inv = [NSInvocation invocationWithMethodSignature:signature];
+    [inv setSelector:@selector(cryptoUsersForVolume:users:)];
+    [inv setReturnValue:&err];
+    [inv setArgument:&targetVol atIndex:2];
+    [inv setArgument:&cryptoUsersPtr atIndex:3];
+    [inv invokeWithTarget:apfs];
+
+    CFRelease(targetVol);
+    CFRelease((__bridge CFTypeRef)apfs);
+    CFRelease(session);
+    CFBundleUnloadExecutable(bundle);
+    CFRelease(bundle);
+
+    if (err != 0) {
+      LOG(ERROR) << "Error calling cryptoUsersForVolume:users:";
+      return;
+    }
+
+    if (cryptoUsers != nullptr) {
+      @autoreleasepool {
+        for (NSString *userUUID in cryptoUsers) {
```

Comment:
Are you sure these are `NSString`s? Should you iterate them as `id` and check if their `NSType` is `NSString` for safety?